### PR TITLE
Fix Critical SonarCloud code smell issues

### PR DESCRIPTION
**Pull request by PeterSuna**
- Rework method cognitive complexity
- Remove duplicate code
- Fix Invoke method(s) only conditionally in LOG.info

### CODE REVIEW

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 1]
```diff
@@ -109,22 +109,22 @@ private void printAugmentations() {
...
-                        final StringBuilder target = new StringBuilder();
-                        for (final QName name : augmentation.getTargetPath().getNodeIdentifiers()) {
-                            target.append('/');
-                            target.append(moduleToPrefix.get(name.getModule()));
-                            target.append(':');
-                            target.append(name.getLocalName());
-                        }
-                        printer.openStatement(Statement.AUGMENT, target.toString());
-                        printOpeningStatement = false;
+                if (isStNotAugmentOrStParentNotEqualsToAugmPath(st, augmentation)) {
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:112]: use positive name for method

that brings possibility to remove `continue` branch

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 2]
```diff
@@ -151,147 +151,172 @@ private void printGroupings(final HashMap<GroupingDefinition, Set<SchemaTree>> t
...
-                doPrintMandatory(schemaNode);
-                doPrintReference(schemaNode);
-                doPrintWhen(schemaNode);
-            }
-            for (final SchemaTree child : tree.getChildren().values()) {
-                doPrintSchema(isPrintingAllowed, child, groupingName, groupingTrees);
-            }
-            if (isPrintingAllowed) {
-                printer.closeStatement();
+        if (!moduleName.equals(schemaNode.getQName().getModule())) {
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:279]: please take positive approach

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 3]
```diff
@@ -55,145 +55,182 @@
...
-                            Collections.emptyList());
-                    ids.remove(ids.size() - 1);
-                }
-                for (Line line : lines) {
-                    final String linesText = line.toString();
-                    LOG.info(linesText);
-                }
-                lines = new ArrayList<>();
+                printLines(getAugmentationNodesLines(singletonListInitializer.getSingletonListWithIncreasedValue(),
+                                                     augNode, removeChoiceQnames));
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:68]: please use continuation indent: `8`, do not align to first parameter

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 4]
```diff
@@ -55,145 +55,182 @@
...
+
+    private void printLines(final List<Line> lines) {
+        for (Line line : lines) {
+            final String linesText = line.toString();
+            LOG.info("{}", linesText);
+        }
+    }
+
+    private List<Line> getNotificationsLines(final SingletonListInitializer singletonListInitializer,
+                                             final Module module, final List<Integer> removeChoiceQnames) {
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:94]: same,
please use continuation indent: `8`, do not align to first parameter

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 5]
```diff
@@ -42,7 +42,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@SuppressFBWarnings("SLF4J_FORMAT_SHOULD_BE_CONST")
+@SuppressFBWarnings("SLF4J_SIGN_ONLY_FORMAT")
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:45]: move these suppresses to specific methods

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 5 above)
Comment (vainyksi) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:45]: I see, those are done in another PR

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java (hunk 1 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java]: done

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java (hunk 2 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java]: done

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 3 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: done

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 4 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: done

**Review by vainyksi — APPROVED** [review]

**Pull request merged**